src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -629,8 +629,31 @@ private bool StartWithCreateProcess(ProcessStartInfo startInfo)
             return ret;
         }
 
-        private static Encoding GetEncoding(int codePage)
+        private static Encoding GetConsoleEncodingOrThrow(ConsoleStreamDirection encodingType)
         {
+            const int INVALID_HANDLE = 6;
+            int codePage;
+            try
+            {
+                codePage = (encodingType == ConsoleStreamDirection.Input) ?
+                    (int)Interop.Kernel32.GetConsoleCP() :
+                    (int)Interop.Kernel32.GetConsoleOutputCP();
+
+                int error = Marshal.GetLastWin32Error();
+                if (error == INVALID_HANDLE)
+                {
+                    throw new PlatformNotSupportedException(SR.ConsoleEncodingNotSupported);
+                }
+                else if (error != 0)
+                {
+                    throw new InvalidOperationException(SR.ConsoleEncodingNotDetected);
+                }
+            }
+            catch (EntryPointNotFoundException)
+            {
+                throw new PlatformNotSupportedException(SR.ConsoleEncodingNotSupported);
```

Comment:
Is throwing the right thing to do for UAP - why not use a default encoding? Otherwise you're saying it is not allowed to redirect input/output at all just because GetconsoleCP fails - is that right? @JeremyKuhne ?